sunkit_image/trace.py

```diff
@@ -0,0 +1,195 @@
+"""
+This module contains functions that will enhance the trace out structures in an
+image.
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+from scipy import interpolate
+
+from sunkit_image.utils import bandpass_filter, erase_loop_in_residual, curvature_radius, initial_direction_finding, loop_add
+
+__all__ = ["occult2"]
+
+
+def occult2(image, nsm1, rmin, lmin, nstruc, nloop, ngap, qthresh1, qthresh2, file=False):
+    """
+    Implements the Oriented Coronal CUrved Loop Tracing (OCCULT-2) algorithm
+    for loop tracing in solar images.
+
+    Parameters
+    ----------
+    image : `numpy.ndarray`
+        Image on which loops are to be detected.
+    nsm1 : `int`
+        Low pass filter boxcar smoothing constant.
+    rmin : `int`
+        The minimum radius of curvature of the loop to be detected in pixels.
+    lmin : `int`
+        The length of the smallest loop to be detected in pixels.
+    nstruc : `int`
+        Maximum limit of traced structures.
+    nloop : `int`
+        Maximum number of detected loops.
+    ngap : `int`
+        Number of pixels in the loop below the flux threshold.
+    qthresh1 : `float`
+        The ratio of image base flux and median flux. All the pixels in the image below `qthresh1 * median` intensity value
+        are made to zero before tracing the loops.
+    qthresh2 : `float`
+        The factor which determines noise in the image. All the intensity values between `qthresh2 * median` are considered
```

Comment:
line is a bit too long

Reply:
But it serves its purpose. I am unable to think of anything that I should remove

Reply:
Again, just linebreak it into two lines.